singletons-base/tests/compile-and-dump/Singletons/PatternMatching.golden

```diff
@@ -594,7 +595,24 @@ Singletons/PatternMatching.hs:(0,0)-(0,0): Splicing declarations
     sX_0123456789876543210 = sPr
     instance SingI (SillySym0 :: (~>) a ()) where
       sing = singFun1 @SillySym0 sSilly
-    instance SingI (Foo2Sym0 :: (~>) (a, b) a) where
+    instance SingI (Foo2Sym0 :: (~>) ((#,#) a b) a) where
       sing = singFun1 @Foo2Sym0 sFoo2
     instance SingI (Foo1Sym0 :: (~>) (a, b) a) where
       sing = singFun1 @Foo1Sym0 sFoo1
+Singletons/PatternMatching.hs:0:0: error: [GHC-83865]
+    • Couldn't match a lifted type with an unlifted type
+      Expected kind ‘Type’,
+        but ‘(#,#) a b’ has kind ‘TYPE
+                                    (GHC.Types.TupleRep
+                                       ((:)
+                                          @GHC.Types.RuntimeRep
+                                          GHC.Types.LiftedRep
+                                          ((:)
+                                             @GHC.Types.RuntimeRep
+                                             GHC.Types.LiftedRep
+                                             ('[] @GHC.Types.RuntimeRep))))’
+    • In a standalone kind signature for ‘Foo2’: (#,#) a b -> a
+   |
+19 | $(singletons [d|
+   |  ^^^^^^^^^^^^^^^...
+
```

Comment:
This error message is unexpected. Upon further inspection, it's GHC fault, as `unboxedTupleTypeName 2` is returning the wrong `Name`, causing `singletons-th` to handle `(#,#)` in an unexpected way. See https://github.com/goldfirere/th-desugar/issues/213, as well as the upstream issue [GHC#24750](https://gitlab.haskell.org/ghc/ghc/-/issues/24750). Happily, that upstream GHC issue has already been fixed in `master`, so once that is backported to GHC 9.10, then this test case should work once more.

Reply:
Happily, GHC 9.10.1 includes a fix for GHC#24750, so this is no longer an issue.